burr/core/application.py

```diff
@@ -1957,6 +1971,29 @@ def _repr_mimebundle_(self, include=None, exclude=None, **kwargs):
         dot = self.visualize(include_conditions=True, include_state=False)
         return dot._repr_mimebundle_(include=include, exclude=exclude, **kwargs)
 
+    def validate_correct_async_use(self):
+        """Validates that the application is meant to run async.
+        This validation is performed in synchronous application methods."""
+
+        # This is a gentle warning for existing users to use the async application
+        if self._adapter_set.async_hooks:
+            logger.warning(
+                "There are asynchronous hooks present in the application that will be ignored. "
+                "Please use async methods to run the application and have them executed. "
+                f"The application has following asynchronous hooks: {self._adapter_set.async_hooks} "
+            )
+
+        # We check that if:
+        # - we have build the application using .abuild()
+        # - we have async hooks present
+        # this application is meant to be run in async mode.
+        if self._builder and self._builder.is_async:
+            raise ValueError(
+                "The application was build with async hooks "
```

Comment:
Make this say "it was built with `abuild`" -- think that's clearer/more actionable